theano/gpuarray/conv_desc.c

```diff
@@ -1,29 +1,57 @@
 #section support_code_apply
 
-int APPLY_SPECIFIC(conv_desc)(PyArrayObject *filt_shp,
+int APPLY_SPECIFIC(conv_desc)(PyArrayObject *filt_shp, PyArrayObject *subsample, PyArrayObject *padding,
+                              PyObject *bmode, PyObject *conv_flag, PyObject *precision_flag,
                               cudnnConvolutionDescriptor_t *desc) {
   cudnnStatus_t err;
-  int pad[3] = {PAD_0, PAD_1, PAD_2};
-  int strides[3] = {SUB_0, SUB_1, SUB_2};
+  int pad[3] = {0,0,0};
+  int strides[3] = {0,0,0};
   int upscale[3] = {1, 1, 1};
+  char *precision_flag;
+  char *conv_mode;
 
-#if BORDER_MODE == 0
+strides[0] = *(npy_int64 *)PyArray_GETPTR1(subsample, 0);
+strides[1] = *(npy_int64 *)PyArray_GETPTR1(subsample, 1);
+strides[2] = *(npy_int64 *)PyArray_GETPTR1(subsample, 2);
+
+if (PyInt_asLong(bmode) == 0L)
```

Comment:
For speed, it would be better to call PyInt_asLong() only once and compare the result many times.